plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/HiveMetastoreBackedDeltaLakeMetastore.java

```diff
@@ -188,6 +224,106 @@ public void renameTable(ConnectorSession session, SchemaTableName from, SchemaTa
         delegate.renameTable(from.getSchemaName(), from.getTableName(), to.getSchemaName(), to.getTableName());
     }
 
+    @Override
+    public void createView(ConnectorSession session, SchemaTableName schemaViewName, ConnectorViewDefinition definition, boolean replace)
```

Comment:
The logic from this method seems to be 1 to 1 copied from `io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog#createView` from `trino-iceberg`.

This applies also for the other methods of the class newly introduced in this PR.

Can you think of a way to avoid such kind of code duplication?

Reply:
The code in Delta and Iceberg is indeed similar but different from the Hive connector as the Hive connector acts upon a `SemiTransactionalhiveMetastore` with slightly different method calls. In Iceberg and Delta we don't have the `SemiTransactionalhiveMetastore`.

To share the code we would need to create a helper class in the Hive connector (the easier part) with all views related operations. This helper class would need access to the `HiveMetastore`  implementation (which will relay to either actual Hive Metastore or `GlueHiveMetastore` (Glue API calls) due to Guice wiring). This works well for Delta. However on Iceberg the implementation is different. It has two factories to create either a `TrinoCatalog` for Hive metastore, which has access to the `HiveMetastore` implementation, either to The Glue catalog which directly calls the Glue API.

We could fix the latter by also using Hive's `GlueHiveMetastore` in the Iceberg`s `TrinoGlueCatalog` in order for the helper class to be used in both cases. There are however also differences between the `TrinoGlueCatalog` and `GlueHiveMetastore` that will need to be investigated and properly tested.

Code reuse is possible but at definitely at a much higher cost also it would probably be better handled in a separate PR. The question arises what we do first: provide the views feature and do the internal refactoring after or first refactor and delay the views feature?

Reply:
I'd go for trying out (as discussed offline) for grouping together the code which would be otherwise repetitive after landing your PR. This seems rather cheap to do and shouldn't add an increased complexity in the code of Delta and Iceberg connectors.

Regarding `TrinoGlueCatalog` I'd suggest to leave it as it is (at least for now).

The main purpose of my comment was DRY (Don't repeat yourself) and not a more complex refactoring (which in any case would not be in the scope of this PR).

Reply:
I have followed your advice and added a `ViewsMetastoreHelper` class, potentially it could also be used for Hive but then we would have to use callbacks. I didn't do that for now.